lstchain/io/io.py

```diff
@@ -209,6 +215,11 @@ def auto_merge_h5files(file_list, output_filename='merged.h5', nodes_keys=None,
     filters
     """
 
+    if type(file_list) != list:
```

Comment:
This is needlessly pedantic. Everything that is an iterable of paths should be taken here. E.g. like this, `auto_merge_h5files(Path().glob("*.h5"))` will throw an error for no reason.

Just do `file_list = list(file_list)` at the start.

Reply:
You're right.
Done.